vault/audited_headers.go

```diff
@@ -0,0 +1,116 @@
+package vault
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/hashicorp/vault/logical"
+)
+
+const (
+	// Key used in the BarrierView to store and retrieve the header config
+	auditedHeadersEntry = "audited_headers"
+	// Path used to create a sub view off of BarrierView
+	auditedHeadersSubPath = "auditedHeadersConfig/"
```

Comment:
Minor nitpick but we don't usually camelcase paths in the backend, but rather use hyphens. So it'd be `sys/audited-headers-config/audited-headers`.